src/lib.rs

```diff
@@ -115,7 +123,86 @@ impl Procyon {
 
         new
     }
-
+    /// save the image
+    pub async fn save(&self, image_path: &Path) -> Result<(), Box<dyn std::error::Error>> {
+        // Spawn in background a webdriver
+        // Currently geckodriver is hardcoded but should support
+        // all webdriver compliant driver.
+        // TODO:updated to all webdriver
+        let mut webdriver_process = Command::new("geckodriver")
+            .args(&["--port", "4444"])
+            .spawn()
+            .expect("webdriver failed to start");
+        // This should also be TODO:updated to all webdriver
+        let mut caps = serde_json::map::Map::new();
+        let opts = serde_json::json!({ "args": ["--headless"]});
```

Comment:
This creates a `serde_json::Value`. If it is necessary to create such a `Value` I guess the macro use is okay.
If you create the `Map` of values and always serialize it as the same string, it may be not. But the call to `Client::with_capabilities` seems to expect a `Map`

Reply:
This is code taken from the example of [fantoccini](https://docs.rs/fantoccini/0.13.0/fantoccini/)